packages/@aws-cdk/aws-ec2/lib/vpc.ts

```diff
@@ -1342,8 +1368,20 @@ class ImportedSubnet extends Resource implements ISubnet, IPublicSubnet, IPrivat
   constructor(scope: Construct, id: string, attrs: SubnetAttributes) {
     super(scope, id);
 
+    if (!attrs.routeTableId) {
+      const ref = Token.isUnresolved(attrs.subnetId)
+        ? `at '${scope.node.path}/${id}'`
+        : `'${attrs.subnetId}'`;
+      // tslint:disable-next-line: max-line-length
+      scope.node.addWarning(`No routeTableId was provided to the subnet ${ref}. Attempting to read it's .routeTable.routeTableId will return null/undefined. (More info: https://github.com/aws/aws-cdk/pull/3171)`);
```

Comment:
Note: For anyone directed to this PR by the `No routeTableId was provided to the subnet` warning message: See this issue: https://github.com/aws/aws-cdk/issues/19786, which about making this warning only appear when necessary.

Reply:
Thanks, but how do I fix the actual warning? I need the route table ids to be pulled back so I can make use of them. Do I have to hard-code them in my config and use `from_attributes( ... routeTableId ...)`?

Reply:
I ended up fetching the subnets and route tables and marrying them via `Subnet.fromSubnetAttributes`. Not ideal, but I'm sure there's a reason...

Reply:
From linked issue, do something like:

```ts
      cdk.Annotations.of(subnet).acknowledgeWarning(
        "@aws-cdk/aws-ec2:noSubnetRouteTableId",
        "Will not read route table ID for subnet",
      );
```